functions.py

```diff
@@ -183,19 +184,23 @@ def get_jenkins_jobs(server, job_search_fields):
 	for i in range(fields_length):
 		fields[i] = fields[i].strip(' ')
 
-	# fetch all jobs from server
-	all_jobs = server.get_jobs()
-
-	# parse out all jobs that do not contain any search field and/or are not OSP10, OSP13, OSP15 or OSP16 jobs
-	relevant_jobs = []
-	supported_versions = ['10', '13', '15', '16', '16.1']
-	for job in all_jobs:
-		job_name = job['name']
-		if any(supported_version in job_name for supported_version in supported_versions):
-			for field in fields:
-				if field in job_name:
+	# check for fields that contain valid regex
+	for field in fields:
+		try:
+			# does the search field contain valid regex?
+			re.compile(field)
+			# fetch all jobs from server that match the given regex
+			all_jobs = server.get_job_info_regex(field)
+
+			# parse out all jobs that do not contain any search field and/or are not OSP10, OSP13, OSP15 or OSP16 jobs
+			supported_versions = ['10', '13', '15', '16', '16.1']
+			for job in all_jobs:
+				job_name = job['name']
+				if any(supported_version in job_name for supported_version in supported_versions):
 					relevant_jobs.append(job)
-					break
+					print(job_name)
+		except re.error as e:
+			print("Error compiling regex: {}".format(e))
 
```

Comment:
We may want a `sys.exit()` here as well; if Jeeves can't search jobs from Jenkins there's not really anywhere else to go from there as far as I see

Reply:
Ah nevermind I see what's going on here; let me rephrase my question - if the given search field is not a regex (think one of our legacy search fields), will `re.compile` fail?

Reply:
No re.compile will not fail. If the given search does not contain valid regex, then the function server.get_job_info_regex(field) will still work by simply looking for jobs containing the string 'field'. 

Also, users may use both regex searches and normal searches at the same time, i.e.
job_search_fields: ^DFG-ceph-rhos, rgw$, unified
This will return:
1. All jobs starting with DFG-ceph-rhos
2. All jobs ending with rgw
3. All jobs containing the string unified

So the only time re.compile fails is when a string has invalid regex, i.e. job_search_fields: '[a-f', ^DFG-ceph-rhos
This will simply fail, output an error in the terminal and move onto '^DFG-ceph-rhos'